### DISCUSSION TITLE
Logging of received beacons

### DISCUSSION THREAD

**Discussion by HarlekinSimplex** [markqvist/Reticulum#73] [Help & Questions] [answered]
Hi all,
yesterday I startet using the beacon feature with thre LoRa interface.
It looks like they are sent (there is a log entry that it has been so) but the receiving nodes show no logentry that it has been received.
Now - How can I switch on logging of incomming beacons so that I see them in the log?
If thats not possible, how can I check wether I receive those beacons properly or not at all?

Cheers
Stephan

**Comment by markqvist** [comment, marked as answer]
Hi Stephan
The transmitted beacons are not logged on the receiving end, since Reticulum itself does not care about them, and they are too obvious a vector to spam peoples logs with random messages.

But you can use [LoRaMon](https://github.com/markqvist/LoRaMon) to inspect the broadcasted beacons.

**Comment by HarlekinSimplex** [comment]
Hi Mark,

 

thats obvious.

 

How ever, if you send those beacons they will be received anyhow, so the harm is already done – literally.

Sending them out without the option to log them on receiving, imho renders the whole feature meaningless.

Such beacons usually are very helpful in setting up connections and to test different antennas.

But for that a logging of the Rx Beacon is necessary.

 

So what about having an option to switch on Rx beacon logging on demand with the default set to off?

Or just log them with log level 7 or like.

 

Actually we are trying to establish a long range LoRa 169 connection with custom made antennas and we need an easy way to find out how good or bad the link is.

Beacons with signal strength logging is the best way to do this, like we can with HAM or CB using aprs beacons (eg. generated by the SW TNC direwolf).

 

Best

Stephan

**Reply by markqvist** [reply]
The beacon feature is primarily intended to allow people to satisfy station ID requirements when used on amateur radio spectrum, so I don't agree that it makes the feature pointless, it is just not intended as a debugging and test tool.

I guess what you really want is to use the `rnprobe` utility, which displays both RSSI and SNR of received probe replies from interfaces that support reporting it.

**Comment by HarlekinSimplex** [comment]
Ah cool.

I’ll check that one out.

And yes - I have the beacon set up to broadcast my CallSign to conform to regulations in place.

😊

 

Von: markqvist ***@***.***> 
Gesendet: Dienstag, 28. Juni 2022 17:19
An: markqvist/Reticulum ***@***.***>
Cc: Stephan Becker ***@***.***>; Author ***@***.***>
Betreff: Re: [markqvist/Reticulum] Logging of received beacons (Discussion #73)

 

The beacon feature is primarily intended to allow people to satisfy station ID requirements when used on amateur radio spectrum, so I don't agree that it makes the feature pointless, it is just not intended as a debugging and test tool.

I guess what you really want is to use the rnprobe utility, which displays both RSSI and SNR of received probe replies from interfaces that support reporting it.

**Comment by markqvist** [comment]
Just be sure to use it against some destination that has it's proof strategy set to `PROVE_ALL`, otherwise `rnprobe` will not get any packets back when a probe is sent. You can use the [echo example]() as a target for `rnprobe`, if you don't have any other destination handy (running it with `echo.py -s` would suffice. You can think of the tool a bit as the `ping` command for IP.
